core/src/test/resources/sql/agg.iq

```diff
@@ -3777,5 +3777,22 @@ select distinct sum(deptno + '1') as deptsum from dept order by 1;
 +---------+
 (1 row)
 
+!ok
+
+# Test cases for [CALCITE-6437] For druid sql JSON_OBJECT() function results in RUNTIME_FAILURE when querying
```

Comment:
This SQL can run success without this PR. And according to this change, I doubt this is a good way to fix this issue in Calcite.